File: go/leetcode/03719.go
package main

func longestBalanced(nums []int) int {
	n := len(nums)
	output := 0

	for i := range n {
		odds := make(map[int]bool, n)
		evens := make(map[int]bool, n)
		for j := i; j < n; j++ {
			if nums[j]%2 == 0 {
				evens[nums[j]] = true
			} else {
				odds[nums[j]] = true
			}

			if len(odds) == len(evens) {
				output = max(output, j-i+1)
			}
		}
	}
	return output
}

File: python/LeetCode/03719.py
from typing import List

class Solution:
    def longestBalanced(self, nums: List[int]) -> int:
        n = len(nums)
        output = 0

        for i in range(n - 1):
            odd = set()
            even = set()

            for j in range(i, n):
                if nums[j] % 2 == 0:
                    even.add(nums[j])
                else:
                    odd.add(nums[j])
                if len(odd) == len(even):
                    output = max(output, j - i + 1)

        return output
